### PR TITLE
Develop upstream sync 241028

### PR DESCRIPTION
**Pull request by ScXfjiang**

### CODE REVIEW

**Comment by ScXfjiang** [comment]
rerun cpu-pycpp please

**Comment by ScXfjiang** [comment]
rerun cpu-pycpp please

**Comment by mmakevic-amd** [comment]
@ScXfjiang I think there might be some unit tests that can be enabled. Try grepping for `#TODO(rocm): weekly sync` to see tests which were disabled in previous weekly syncs.

For example, I think `pjrt_c_api_gpu_test` is passing on upstream now, so that can be re-enabled

**Comment by ScXfjiang** [comment]
> @ScXfjiang I think there might be some unit tests that can be enabled. Try grepping for `#TODO(rocm): weekly sync` to see tests which were disabled in previous weekly syncs.
> 
> For example, I think `pjrt_c_api_gpu_test` is passing on upstream now, so that can be re-enabled

It seems it is still failing:
image

**Comment by mmakevic-amd** [comment]
> > @ScXfjiang I think there might be some unit tests that can be enabled. Try grepping for `#TODO(rocm): weekly sync` to see tests which were disabled in previous weekly syncs.
> > For example, I think `pjrt_c_api_gpu_test` is passing on upstream now, so that can be re-enabled
> 
> It seems it is still failing: image

Weird, I can see it was passing yesterday on XLA nightly (http://[ml-ci.amd.com:21096/job/OpenXLA-Internal/job/XLA-nightly/661/consoleText](http://ml-ci.amd.com:21096/job/OpenXLA-Internal/job/XLA-nightly/661/consoleText))

**Comment by ScXfjiang** [comment]
rerun cpu-pycpp please

**Comment by ScXfjiang** [comment]
rerun gpu-pycpp please

File: third_party/xla/xla/service/gpu/fusions/triton/dot_algorithms_test.cc [hunk 1]
```diff
@@ -1146,6 +1146,9 @@ CHECK-NOT: mma.sync.aligned.{{.*}}.row.col.f32.tf32.tf32.f32
 }
 
 TEST_F(TritonAlgorithmTest, Algorithm_BF16_BF16_F32_X3) {
+  if (std::holds_alternative<se::RocmComputeCapability>(GpuComputeComp())) {
+    GTEST_SKIP() << "Triton currently disabled on ROCM.";
+  }
```
Comment (i-chaochen) [inline third_party/xla/xla/service/gpu/fusions/triton/dot_algorithms_test.cc:1152]: when you skip one subtest, could you also leave a comment to indicate the date like this, please? https://github.com/ROCm/tensorflow-upstream/pull/2704/commits/f79915189d42c069fbc993874587bae28579e339 

in addition, could you please add this requirement to here as well? https://github.com/ROCm/tensorflow-upstream/blob/develop-upstream/rocm_docs/SYNC_UPSTREAM.md

**Comment by ScXfjiang** [comment]
> > > @ScXfjiang I think there might be some unit tests that can be enabled. Try grepping for `#TODO(rocm): weekly sync` to see tests which were disabled in previous weekly syncs.
> > > For example, I think `pjrt_c_api_gpu_test` is passing on upstream now, so that can be re-enabled
> > 
> > 
> > It seems it is still failing: image
> 
> Weird, I can see it was passing yesterday on XLA nightly (http://[ml-ci.amd.com:21096/job/OpenXLA-Internal/job/XLA-nightly/661/consoleText](http://ml-ci.amd.com:21096/job/OpenXLA-Internal/job/XLA-nightly/661/consoleText))

The CI is currently running. I submitted another PR to enable the test to see if it can pass: https://github.com/ROCm/tensorflow-upstream/pull/2741

File: third_party/xla/xla/service/gpu/fusions/triton/dot_algorithms_test.cc (hunk 1 above)
Comment (ScXfjiang) [reply third_party/xla/xla/service/gpu/fusions/triton/dot_algorithms_test.cc:1152]: Done. The PR is here: https://github.com/ROCm/tensorflow-upstream/pull/2742

**Comment by hsharsha** [comment]
https://github.com/bazelbuild/bazel/issues/24173

**Review by mmakevic-amd — APPROVED** [review]

**Pull request merged**